packages/vite/src/node/plugins/resolve.ts

```diff
@@ -831,10 +831,10 @@ export function resolvePackageEntry(
         return resolvedEntryPoint
       }
     }
-    packageEntryFailure(id)
   } catch (e) {
     packageEntryFailure(id, e.message)
   }
+  packageEntryFailure(id)
```

Comment:
How does this work? When there is an exception, won't `packageEntryFailure` be called twice after this change? And if there isn't an exception, isn't it the same to place the function at the end of the try block or after it 🤔

Reply:
`packageEntryFailure` is essentially a `throw new Error()`.

So the previous behavior is:

- case (a): no package entry is found
    1. `packageEntryFailure(id)` is called in L834
    2. an error is thrown inside that
    3. that error is caught by `catch` in L835
    4. `packageEntryFailure(id, e)` is called in L836
    5. an error is thrown inside that and gets out of this function
- case (b): error happened while resolving package entry
    1. error happens inside the `try` block
    2. that error is caught by `catch` in L835
    3. `packageEntryFailure(id, e)` is called in L836
    4. an error is thrown inside that and gets out of this function

the PR behavior is:

- case (a): no package entry is found
    1. `packageEntryFailure(id)` is called in L837
    2. an error is thrown inside that and gets out of this function
- case (b): error happened while resolving package entry
    - this is the same with the current behavior

Reply:
Thanks for the explanation! I overlooked the throw error as a log only inside that function. case (b) would still log twice then, no?
Maybe it could be something like:
```
  let errorMessage?: string  
  try {
    // ...
  } catch (e) {
    errorMessage = e.message
  } 
  packageEntryFailure(id, errorMessage)
```